src/net_processing.cpp

```diff
@@ -1643,10 +1640,11 @@ void PeerManagerImpl::NewPoWValidBlock(const CBlockIndex *pindex, const std::sha
 
     if (!DeploymentActiveAt(*pindex, m_chainman, Consensus::DEPLOYMENT_SEGWIT)) return;
 
-    uint256 hashBlock(pblock->GetHash());
+    auto pcmpctblock = std::make_shared<const CBlockHeaderAndShortTxIDs>(*pblock);
```

Comment:
It looks like we don't call `NewPoWValidBlock` while we are in IBD (see [here](https://github.com/bitcoin/bitcoin/blob/84bf31f8e9842cb5f613151dc4c5a0ffa67a3a03/src/validation.cpp#L3759-L3760)), so 40ae3107956e2425c4a3cf0cb3a4a2d6d952894c won't really affect IBD speed. Consider dropping that commit or changing the commit message if there is another motivation for the change.

Reply:
Good catch! I've dropped the commit (this function could be tidied up a bit in terms of commenting/variable instantiation, but that doesn't seem strictly related to this PR).